include/seqan3/range/views/interleave.hpp

```diff
@@ -74,10 +76,18 @@ class view_interleave : public std::ranges::view_interface<view_interleave<urng_
                                                          std::ranges::range_reference_t<inserted_rng_t const>>, void>;
     //!\brief The value_type (which equals the reference_type with any references removed).
     using value_type        = std::ranges::range_value_t<urng_t>;
-    //!\brief This resolves to range_type::size_type as the underlying range is guaranteed to be Sized.
-    using size_type         = size_type_t<urng_t>;
     //!\brief A signed integer type, usually std::ptrdiff_t.
     using difference_type   = std::ranges::range_difference_t<urng_t>;
+    //!\}
+
+public:
+    //!\brief Befriend the following class s.t. iterator and const_iterator can be defined for this type.
+    template <typename parent_type, typename crtp_base>
+    friend class detail::random_access_iterator_base;
```

Comment:
Move the friend declaration to the private section, it can be hidden from the user as it is an implementation detail. 
Also check the documentation. I am not sure if the brief shows in friends?

Reply:
Moved and checked: the brief shows.